Libraries/LibJS/Runtime/IndexedProperties.cpp

```diff
@@ -266,6 +261,21 @@ ValueAndAttributes IndexedPropertyIterator::value_and_attributes(Object* this_ob
     return {};
 }
 
+void IndexedPropertyIterator::skip_empty_indices()
+{
+    auto indices = m_indexed_properties.indices();
```

Comment:
This feels like a heavy thing to do on every iterator increment.
It's okay for now, but maybe later we should consider caching the indices in the iterator.

Reply:
Yep there's definitely still room for improvements here